python/lsst/dax/imgserv/getimage/coadd.py

```diff
@@ -0,0 +1,177 @@
+#
+# LSST Data Management System
+# Copyright 2008, 2009, 2010, 2011, 2012 LSST Corporation.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+import math
+
+import lsst.log as log
+import lsst.pex.config as pexConfig
+import lsst.afw.image as afwImage
+import lsst.afw.math as afwMath
+import lsst.coadd.utils as coadd_utils
+
+class CoaddConfig(pexConfig.Config):
+    """Config for Coadd
+    """
+    badMaskPlanes = pexConfig.ListField(
+        dtype=str,
+        doc="mask planes that, if set, the associated pixel should not be included in the coadd",
+        default=("NO_DATA", "SAT"),
+    )
+
+
+class Coadd:
+    """Coadd by weighted addition
+
+    This class may be subclassed to implement other coadd techniques.
+    Typically this is done by overriding addExposure.
+    """
+    ConfigClass = CoaddConfig
+
+    def __init__(self, bbox, wcs, badMaskPlanes, logName="coadd.utils.Coadd"):
+        """Create a coadd
+
+        @param[in] bbox: bounding box of coadd Exposure with respect to parent (lsst.afw.geom.Box2I):
```

Comment:
Please use numpydoc not doxygen for new docstrings.

Reply:
Converting all strings to numpydoc format in that file.